src/main/java/edu/harvard/iq/dataverse/util/SignpostingResources.java

```diff
@@ -0,0 +1,337 @@
+    String defaultFileTypeValue;
+    int maxAuthors;
+    int maxItems;
+
+    public SignpostingResources(SystemConfig systemConfig, DatasetVersion workingDatasetVersion, String jsonSetting) {
+        this.systemConfig = systemConfig;
+        this.workingDatasetVersion = workingDatasetVersion;
+        if (jsonSetting == null) {
+            jsonSetting = BundleUtil.getStringFromBundle("signposting.configuration.SignpostingConf");
+        }
+        JsonReader jsonReader = Json.createReader(new StringReader(jsonSetting));
+        JsonObject spJsonSetting = jsonReader.readObject();
+        jsonReader.close();
+//        licJsonObj = spJsonSetting.getJsonObject("license");
+        describedByJsonObj = spJsonSetting.getJsonObject("describedby");
+        useDefaultFileType = spJsonSetting.getBoolean("useDefaultFileType", true);
+        defaultFileTypeValue = spJsonSetting.getString("defaultFileTypeValue", "https://schema.org/Dataset");
+        maxAuthors = spJsonSetting.getInt("maxAuthors", 5);
+        maxItems = spJsonSetting.getInt("maxItems", 5);
+    }
+
+    /**
+     * Get identifier schema for each author
+     * <p>
+     * For example:
+     * if author has VIAF
+     * Link: <http://viaf.org/viaf/:id/>; rel="author"
+     *
+     * @param datasetAuthors list of all DatasetAuthor object
+     * @return all the non empty author links in a string
+     */
+    private String getAuthors(List<DatasetAuthor> datasetAuthors) {
+        String singleAuthorString;
+        String identifierSchema = "";
+        int visibleAuthorCounter = 0;
+//        if (datasetAuthors.size() > maxAuthors) {return "";}
+        for (DatasetAuthor da : datasetAuthors) {
+            logger.info(String.format(
+                    "idtype: %s; idvalue: %s, affiliation: %s; identifierUrl: %s",
+                    da.getIdType(),
+                    da.getIdValue(),
+                    da.getAffiliation(),
+                    da.getIdentifierAsUrl()
+            ));
+
+            String authorURL = "";
+            authorURL = getAuthorUrl(da);
+            if (!Objects.equals(authorURL, "")) {
+                visibleAuthorCounter++;
+                // return empty if number of visible author more than max allowed
+                if (visibleAuthorCounter >= maxAuthors) return "";
+                singleAuthorString = "<" + authorURL + ">;rel=\"author\"";
+                if (Objects.equals(identifierSchema, "")) {
+                    identifierSchema = singleAuthorString;
+                } else {
+                    identifierSchema = String.join(",", identifierSchema, singleAuthorString);
+                }
+            }
+        }
+
+        logger.info(String.format("identifierSchema: %s", identifierSchema));
+        return identifierSchema;
+    }
+
+    /**
+     * Get key, values of signposting items and return as string
+     *
+     * @return comma delimited string
+     */
+    public String getLinks() {
```

Comment:
A similar question to authors - this is assembling the same info as in the linkset except that this a) results in a string and b) enforces the max limits? Seems like basing this off the json output with the max limits as params would avoid duplicating the code to retrieve this info?